### DISCUSSION TITLE
Crossfade switch from HA?

### DISCUSSION THREAD

**Discussion by anha7196** [music-assistant/support#3042] [Q&A] [answered]
Hi!

I want to play a looping sleep sound track on my daughters Sonos player, triggered by pressing play on the Sonos device within a set time period. The automation works fine, but i need the track to loop and crossfade so that there is no jarring pauses in the sound. Looping works fine but i can't fine any way of enabling crossfade from within Home Assistant. Am i missing it or is not possible as of now?

If i start the track from the MA interface and enable crossfade there it works fine.

Thanks!

/Anders

**Comment by OzGav** [comment]
There is no HA action which enables crossfade. Crossfade remains enabled on the player once you set it so that is an option.

**Reply by anha7196** [reply]
Thanks for the quick reply! Now I can stop looking for it at least. 😃 
I need to investigate some more though, it seems that the crossfade reverts back quite often for me.

**Reply by OzGav** [reply]
It should be persistent but restart MA and see if it reverts then

**Reply by anha7196** [reply]
Thanks, I'll try as soon i have time to tinker!

**Comment by OzGav** [comment, marked as answer]
From MA 2.4 there is an API (not yet documented) that can be used to change the crossfade state of the player. Get the player_id from the top of the player settings. Not this is not guaranteed to continue working as this API is not for general user consumption.

```
rest_command:
  ma_set_player_settings:
    url: http://localhost:8095/api
    method: POST
    headers:
      accept: "application/json, text/html"
    payload: >
      {
        "message_id": "1",
        "command": "config/players/save",
        "args": {
          "player_id": "b8:27:eb:8a:b8:8e",
          "values": {
            "crossfade": true
          }
        }
      }
    content_type:  'application/json; charset=utf-8'
```

**Comment by OzGav** [comment]
There is now API documentation for the 2.7.0 beta and later versions of Music Assistant. The API documentation is automatically generated and available at http://your_ma_server_ip:8095/api-docs
